src/modules/commander/esc_calibration.cpp

```diff
@@ -123,8 +110,15 @@ int do_esc_calibration(orb_advert_t *mavlink_log_pub, struct actuator_armed_s* a
 	}
 
 	// Make sure battery is disconnected
-	orb_copy(ORB_ID(battery_status), batt_sub, &battery);
-	if (battery.voltage_filtered_v > 3.0f) {
+	if (orb_copy(ORB_ID(battery_status), batt_sub, &battery) == PX4_OK) {
+
+		// battery is not connected if voltage is lower than 3V and we have a recent battery measurement
+		if (battery.voltage_filtered_v < 3.0f && (hrt_absolute_time() - battery.timestamp < 500*1000)) {
```

Comment:
You might want to start using hrt_elapsed_time() instead. There's been talk of adding additional checks within. https://github.com/PX4/Firmware/pull/8912
 
500*1000 -> 500_ms literal

Reply:
There's also a battery_status.connected field. Wouldn't that be better than checking a magic voltage?